generic3g/OuterMetaComponent/initialize_read_restart.F90

```diff
@@ -0,0 +1,49 @@
+#include "MAPL_Generic.h"
+
+submodule (mapl3g_OuterMetaComponent) initialize_read_restart_smod
+
+   use mapl3g_GenericPhases
+   use mapl_ErrorHandling
+   use mapl3g_MultiState
+   use mapl3g_RestartHandler, only: RestartHandler
+
+   implicit none
+
+contains
+
+   module recursive subroutine initialize_read_restart(this, unusable, rc)
+      class(OuterMetaComponent), intent(inout) :: this
+      ! optional arguments
+      class(KE), optional, intent(in) :: unusable
+      integer, optional, intent(out) :: rc
+
+      character(*), parameter :: PHASE_NAME = 'GENERIC::INIT_READ_RESTART'
+      type(GriddedComponentDriver), pointer :: driver
+      character(len=:), allocatable :: name
+      type(MultiState) :: states
+      type(ESMF_State) :: import_state, internal_state
+      type(RestartHandler) :: restart_handler
+      integer :: status
+
+      driver => this%get_user_gc_driver()
```

Comment:
This routine lacks recursion over children.

Reply:
The recursion is on line 41 - in the `if` block that checks if the name is `HIST` https://github.com/GEOS-ESM/MAPL/blob/e8e1d88b3302481a4333464240770fae2eb1971f/generic3g/OuterMetaComponent/initialize_read_restart.F90#L40-L42

Reply:
Ah - ok.   The original implementation made the recursion look like the exceptional case.

  The alternative (if HIST was still needed as an exception) would be:
```
_RETURN_IF(name == "HIST")
call recurse(...)
```

Am starting to wonder if `recurse` should be made a method. ...